### PR TITLE
Fixed sprand for dimensions of unusual integer type

### LINKED ISSUE #30502: wrong ouput of sprand with dimensions of unusual integer type
```
julia> m,n=UInt8(16),UInt8(16);

julia> m,n=UInt8(16),UInt8(16); sprand(m,n,1.0)
16×16 SparseMatrixCSC{Float64,Int64} with 0 stored entries
```

### PR DESCRIPTION
**Pull request by raghav9-97**
Fixes #30502.

### CODE REVIEW

**Comment by abraunst** [comment]
You should also fix `sprand(m, n, p, rfn)` (the function right below); otherwise `sprand(UInt8(16), UInt8(16), 1.0, ones)` has the same problem.

**Comment by raghav9-97** [comment]
Please re review this PR.

**Comment by ViralBShah** [comment]
@raghav9-97 PRs do get reviewed as you submit them (unless you put WIP in the title). In general, you want to bump it up for review if it has not received any attention in about 3-4 days. Hope that helps.

File: stdlib/SparseArrays/test/sparse.jl [hunk 1]
```diff
@@ -339,6 +339,12 @@ end
     end
 end
 
+@testset "Issue #30502" begin
+    @test sprand(UInt8(16), UInt8(16), 1.0) != spzeros(UInt8(16), UInt8(16))
+    @test sprand(Int8(16), Int8(16), 1.0) != spzeros(Int8(16), Int8(16))
+    @test sprand(UInt8(16), UInt8(16), 1.0, ones) != spzeros(UInt8(16), UInt8(16))
+end
```
Comment (ViralBShah) [inline stdlib/SparseArrays/test/sparse.jl]: Can we have a test that checks correctness of the intended answer with `==`? Also, only one test would be enough here.

**Review by ViralBShah — CHANGES_REQUESTED** [review]

**Comment by raghav9-97** [comment]
Is this PR now ready to be merged?

**Pull request merged**